server/internal/usecase/repo/Item.go

```diff
@@ -18,6 +18,7 @@ type Item interface {
 	FindByIDs(context.Context, id.ItemIDList) (item.List, error)
 	FindAllVersionsByID(context.Context, id.ItemID) ([]*version.Value[*item.Item], error)
 	IsArchived(context.Context, id.ItemID) (bool, error)
+	FindByModelAndValue(context.Context, id.ModelID, any) (*item.Item, error)
```

Comment:
```suggestion
	FindByModelAndValue(context.Context, id.ModelID, []id.FieldID, []any) (*item.Item, error)
```

Or

```suggestion
	FindByModelAndValue(context.Context, id.ModelID, []struct{Field id.FieldID; Value any}) (*item.Item, error)
```

A more rigorous examination using field ID/value combinations should be used to detect appropriate items. With a little ingenuity, unnecessary data exchange can be reduced.